engine/crates/wasi-component-loader/src/error.rs

```diff
@@ -1,5 +1,38 @@
 pub mod guest;
 
+/// The error type from a WASI call from the gateway.
+#[derive(Debug, thiserror::Error)]
+pub enum GatewayError {
```

Comment:
This would mean a boxed error trait, right? And then downcasting here?

https://github.com/grafbase/grafbase/blob/gb-7872/engine/crates/runtime-local/src/hooks.rs#L283-L296

Reply:
I'm not sure to follow there you would call `run_and_measure::<Error<ErrorResponse>>(...)`, no?

Reply:
Achso, yeah!

Reply:
This became very annoying thing to change, I ended up fixing our pool in the runtime-local which was very unhappy about the new error type parameter. Let's see when we are in a moment when there's not million issues to do in the backlog... I'll leave this how it is now without spending too much time...